modules/rs_grading.py

```diff
@@ -138,10 +140,12 @@ def _scorable_useinfos(course_name, sid, div_id, points, deadline, event_filter=
         query = query & (db.useinfo.timestamp < deadline)
     if practice_start_time:
         query = query & (db.useinfo.timestamp >= practice_start_time)
+        query = query & (db.useinfo.timestamp <= now)
```

Comment:
This should be inside an if statement:
```
if now:
    query = query & (db.useinfor.timestamp <= now)
```

And similarly for the rest of the functions where you're adding this.

Reply:
I added the if statements, but after checking the existence of `practice_start_time`, I already know that `now` also exists.